.github/workflows/blocks-playwright.yml

```diff
@@ -58,19 +69,74 @@ jobs:
                   path: plugins/woocommerce-blocks/tests/e2e/artifacts/test-results
                   if-no-files-found: ignore
 
+            - name: Archive flaky test reports
+              uses: actions/upload-artifact@v4
+              if: ${{ !cancelled() }}
+              with:
+                  name: flaky-tests-report-shard-${{ matrix.shardIndex }}
+                  path: plugins/woocommerce-blocks/flaky-tests
+                  if-no-files-found: ignore
+
     merge-artifacts:
         # Merges all artifacts from all shards into a single zip and
         # deletes the parts. In case of a rerun, artifacts from the
-        # previous run will be retained by merging them with the new ones. 
+        # previous run will be retained by merging them with the new ones.
         name: Merge Artifacts
         if: ${{ !cancelled() }}
         needs: [blocks-playwright-tests]
         runs-on: ubuntu-latest
+        outputs:
+            has-flaky-test-report: ${{ !!steps.merge-flaky-tests-reports.outputs.artifact-id }}
         steps:
-            - name: Merge Artifacts
+            - name: Merge failures artifacts
               uses: actions/upload-artifact/merge@v4
               # Don't fail the job if there aren't any artifacts to merge.
               continue-on-error: true
               with:
                   name: failures-artifacts
+                  pattern: failures-artifacts*
                   delete-merged: true
+
+            - name: Merge flaky tests reports
+              id: merge-flaky-tests-reports
+              uses: actions/upload-artifact/merge@v4
+              continue-on-error: true
+              with:
+                  name: flaky-tests-report
+                  pattern: flaky-tests-report*
+                  delete-merged: true
+
+    create-github-issues-for-flaky-tests:
+        name: Create GitHub issues for flaky tests
+        needs: [merge-artifacts]
+        if: ${{ !cancelled() && needs.merge-artifacts.outputs.has-flaky-test-report == 'true' }}
+        runs-on: ubuntu-latest
+        permissions:
+            contents: read
+            issues: write
+        steps:
+            # We are using @wordpress/report-flaky-tests package from Gutenberg repo
+            # to create GitHub issues for flaky tests.
+            - uses: actions/checkout@v4
+              with:
+                  repository: WordPress/gutenberg
+                  ref: trunk
+
+            - uses: actions/download-artifact@v4
+              with:
+                  name: flaky-tests-report
+                  path: flaky-tests
+
+            - name: Setup Node.js and install dependencies
+              uses: ./.github/setup-node
+
+            - name: Npm build
+              # TODO: We don't have to build the entire project, just the action itself.
+              run: npm run build:packages
```

Comment:
I can see that's also the case in Gutenberg. What's actually the blocking part here? Why can't we build the package alone?

Reply:
AFAIU Packages build is done using [build.js](https://github.com/WordPress/gutenberg/blob/13717c6d83b3548fc967f1006cffb09093e1faec/bin/packages/build.js) file. This file doesn't support building a specific package, therefore we are building all the packages. Am I missing something? 🤔

Reply:
I don't really know, but it sounds like something that would be great to improve - building one package is much faster than building all the packages! 😅 It's an upstream task though, so not a blocker at all!

Reply:
Yeah, once it's done upstream, we can make the same changes in our repo. We may consider contributing upstream too.